src/Agent.Listener/Configuration/ConfigurationProvider.cs

```diff
@@ -215,6 +215,7 @@ public virtual async Task DeleteAgentAsync(AgentSettings agentSettings)
             }
         }
 
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Maintainability", "CA2000:Dispose objects before losing scope", MessageId = "CreateConnection")]
```

Comment:
I guess generally we do not care about tests

Reply:
well, this is weird, because the method is call Test, but it establishes the connection and then it's used later assuming the Test "passed"